src/core/index_meta.rs

```diff
@@ -112,10 +112,16 @@ impl SegmentMeta {
             .load(std::sync::atomic::Ordering::Relaxed)
         {
             SegmentComponent::iterator()
+                .filter(|comp| {
```

Comment:
Yeah as you spotted, the comment is not accurate.

It contains a superset of the files that are useful for the index.
Some of the file in the returned list may not exist.

You spotted the delete file, but it is not the only one.
One thing you can do is to use the list of managed files and do
`managed files - list_files()`